python/tests/integration/arcticdb/version_store/test_basic_version_store.py

```diff
@@ -1750,7 +1749,7 @@ def test_diff_long_stream_descriptor_mismatch(lmdb_version_store, method, num):
             lib.update("x", pd.DataFrame(bad_row, index=dr), date_range=dr)
         assert False, "should throw"
     except StreamDescriptorMismatch as e:
-        print(e)
+        assert not isinstance(e, _ArcticLegacyCompatibilityException)
```

Comment:
Wait, so `StreamDescriptorMismatch` is actually outside the class hierarchy?

I did it by accident. All ArcticDB exceptions should be catch-able via a common type.

Reply:
Cool. For `StreamDescriptorMismatch` we can reasonably safely put it in the hierarchy since it's newer. For `NoDataFoundException` though, I think it's too late, we could break existing programs by putting it in the hierarchy.